File: src/test/java/de/wasenweg/alfred/fixtures/VolumeFixtures.java
package de.wasenweg.alfred.fixtures;

import de.wasenweg.alfred.volumes.Volume;

public final class VolumeFixtures {

  public static final Volume VOLUME_1 = Volume.builder()
      .name(ComicFixtures.COMIC_V1_1.getVolume())
      .series(ComicFixtures.COMIC_V1_1.getSeries())
      .publisher(ComicFixtures.COMIC_V1_1.getPublisher())
      .issueCount(3)
      .read(false)
      .readCount(0)
      .firstComicId(ComicFixtures.COMIC_V1_1.getId())
      .build();

  private VolumeFixtures() {
    throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
  }
}

File: ui/src/testing/volume.fixtures.ts
import { Volume } from 'src/app/volume';

export class VolumeFixtures {

  public static get volume(): Volume {
    const volume: Volume = {} as Volume;
    volume.name = '1999';
    volume.series = 'Batgirl';
    volume.publisher = 'DC Comics';
    volume.read = false;
    volume.issueCount = 10;
    volume.readCount = 0;
    volume.firstComicId = '1234';

    return volume;
  }

  public static get volumes(): Volume[] {
    return [
      this.volume,
      Object.assign(this.volume, { name: '2003' }),
      Object.assign(this.volume, { name: '2008' }),
      Object.assign(this.volume, { name: '2009' })
    ];
  }
}
